api-registration/server/index.js

```diff
@@ -0,0 +1,66 @@
+require('dotenv/config');
+const pg = require('pg');
+const argon2 = require('argon2'); // eslint-disable-line
+const express = require('express');
+const ClientError = require('./client-error');
+const errorMiddleware = require('./error-middleware');
+
+const db = new pg.Pool({ // eslint-disable-line
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.post('/api/auth/sign-up', (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
+    throw new ClientError(400, 'username and password are required fields');
+  }
+
+  /* your code starts here */
+
+  /**
+   * Hash the user's password with `argon2.hash()`
+   * Then, 😉
+   *   Insert the user's "username" and "hashedPassword" into the "users" table.
+   *   Then, 😉
+   *     Respond to the client with a 201 status code and the new user's "userId", "username", and "createdAt" timestamp.
+   *   Catch any errors.
+   * Catch any errors.
+   *
+   * Hint: Insert statements can include a `returning` clause to retrieve the insterted row(s).
+   */
+  argon2.hash(password)
+    .then(hashedPassword => {
+      const sql = `
+        INSERT INTO "users" ("username", "hashedPassword")
+        VALUES ($1, $2)
+        RETURNING *
+      `;
+      db.query(sql, [username, hashedPassword])
+        .then(dbResponse => {
+          const newUser = dbResponse.rows[0];
+          res.status(201).json({
+            userId: newUser.userId,
+            username: newUser.username,
+            createdAt: newUser.createdAt
+          });
```

Comment:
This will work to send back the correct information back to the client, but an easier way would be to control what is returned from the SQL statement by specifying the specific desired columns in the `returning` clause of the SQL statement. For example:

```SQL
returning "userId", "username", "createdAt"
```

Reply:
Got it! Thank you!